### PR TITLE
Refactoring: Config class + modules flow UML + minors

### PR DESCRIPTION
**Pull request by konbraphat51**
## What needs occurred
#38 
#42
#44

## What you changed (What reviewers have to see)
- Create & Use Config class
- Make UML
- minors

## Description
Nothing special to say

## Check list
- [x] Code styles confirmed
- [x] Test confirmed

### CODE REVIEW

**Comment by konbraphat51** [comment]
@SuperHotDogCat 
review & merge #42 first, then merge it into this.

But you can review this now, so your work later be less

File: tests/Auto/Animator/ImageCreator_test.py [hunk 1]
```diff
@@ -26,11 +25,7 @@ def test_imagecreator():
     allocation_in_frame = AllocationInFrame()
     allocation_in_frame.words = {"word": (30, (50, 50))}  # dictionary
     position_in_frames.add("2023_04_01", allocation_in_frame)
-    create_images(
-        position_in_frames,
-        image_size=(100, 100),
-        font_path=DEFAULT_ENG_FONT_PATH,
-    )
+    create_images(position_in_frames, Config())
     test_path = os.path.join(DEFAULT_OUTPUT_PATH, "*.png")
```
Comment (SuperHotDogCat) [inline tests/Auto/Animator/ImageCreator_test.py:29]: Although I wrote this line, change this line to """os.path.join(DEFAULT_OUTPUT_PATH, "2023_04_01.png")"""

File: tests/Auto/Animator/ImageCreator_test.py [hunk 2]
```diff
@@ -26,11 +25,7 @@ def test_imagecreator():
...
     assert len(glob.glob(test_path)) != 0
```
Comment (SuperHotDogCat) [inline tests/Auto/Animator/ImageCreator_test.py:30]: Although I wrote this line, change this line to """os.path.isfile(test_path)"""

**Comment by konbraphat51** [comment]
Fixed in [refac_merged branch](https://github.com/konbraphat51/AnimatedWordCloud/tree/refac_merged)
Waiting for #42 approved

**Comment by konbraphat51** [comment]
@SuperHotDogCat Ok, next, this

File: AnimatedWordCloud/Animator/AllocationCalculator/StaticAllocationCalculator/StaticAllocationCalculator.py [hunk 3]
```diff
@@ -79,13 +63,15 @@ def allocate(
...
     # calculate allocation by selected strategy
-    if strategy == "magnetic":
+    if config.allocation_strategy == "magnetic":
         allocator = MagneticAllocation(
-            image_width, image_height, image_division
+            config.image_width, config.image_height, config.image_division
         )
         return allocator.allocate(words, allocation_before)
     else:
-        raise ValueError("Unknown strategy: {}".format(strategy))
```
Comment (SuperHotDogCat) [inline AnimatedWordCloud/Animator/AllocationCalculator/StaticAllocationCalculator/StaticAllocationCalculator.py:88]: Do you plan to use any other arguments than the magnetic strategy? If you do not, please skip this review.

Comment (SuperHotDogCat) [reply AnimatedWordCloud/Animator/AllocationCalculator/StaticAllocationCalculator/StaticAllocationCalculator.py:88]: For example,
config.allocation_strategy: str | function
if config.allocation_strategy == "magnetic":
   // allocate by the magnetic strategy
elif type(config.allocation_strategy) == function:
  // allocate by your own strategy
else:
  raise error

Comment (SuperHotDogCat) [reply AnimatedWordCloud/Animator/AllocationCalculator/StaticAllocationCalculator/StaticAllocationCalculator.py:88]: This change could be in the future version.

**Review by SuperHotDogCat — COMMENTED** [review]
I remembered that matplotlib has a function to display available fonts.
Maybe it would be good to have the function in this library if possible.

File: AnimatedWordCloud/Animator/AllocationCalculator/StaticAllocationCalculator/StaticAllocationCalculator.py (hunk 3 above)
Comment (konbraphat51) [reply AnimatedWordCloud/Animator/AllocationCalculator/StaticAllocationCalculator/StaticAllocationCalculator.py:88]: There could be another strategy, so I made an option for backward compatibility

**Comment by konbraphat51** [comment]
About fonts, I will leave an issue #48, let's focus on this scope for now

**Review by SuperHotDogCat — APPROVED** [review]
LGTM

**Pull request merged**